httpclient/src/main/java/feign/httpclient/ApacheHttpClient.java

```diff
@@ -153,13 +153,19 @@ HttpUriRequest toHttpUriRequest(Request request, Request.Options options) throws
   private ContentType getContentType(Request request) {
     ContentType contentType = ContentType.DEFAULT_TEXT;
     for (Map.Entry<String, Collection<String>> entry : request.headers().entrySet())
-    if (entry.getKey().equalsIgnoreCase("Content-Type")) {
-      Collection values = entry.getValue();
-      if (values != null && !values.isEmpty()) {
-        contentType = ContentType.create(entry.getValue().iterator().next(), request.charset());
-        break;
+      if (entry.getKey().equalsIgnoreCase("Content-Type")) {
+        Collection<String> values = entry.getValue();
+        if (values != null && !values.isEmpty()) {
+          String mimeType = values.iterator().next();
```

Comment:
yeap if there's a feature already to do that, by all means.. use it!

On Wed, Sep 7, 2016 at 11:04 PM, Andrey Stolyarchuk <
notifications@github.com> wrote:

> In httpclient/src/main/java/feign/httpclient/ApacheHttpClient.java
> <https://github.com/OpenFeign/feign/pull/453#discussion_r77840620>:
>
> > @@ -153,13 +153,19 @@ HttpUriRequest toHttpUriRequest(Request request, Request.Options options) throws
> >    private ContentType getContentType(Request request) {
> >      ContentType contentType = ContentType.DEFAULT_TEXT;
> >      for (Map.Entry<String, Collection<String>> entry : request.headers().entrySet())
> > -    if (entry.getKey().equalsIgnoreCase("Content-Type")) {
> > -      Collection values = entry.getValue();
> > -      if (values != null && !values.isEmpty()) {
> > -        contentType = ContentType.create(entry.getValue().iterator().next(), request.charset());
> > -        break;
> > +      if (entry.getKey().equalsIgnoreCase("Content-Type")) {
> > +        Collection<String> values = entry.getValue();
> > +        if (values != null && !values.isEmpty()) {
> > +          String mimeType = values.iterator().next();
>
> I agree, in this case you are doing the same as already done in apache
> http client code. Would not it be easer just to use ContentType.parse
> instead of create?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/OpenFeign/feign/pull/453/files/e4cd7a34ee6143b6b50adf85e7273119a8120ccc#r77840620>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAD61y_Mm4_zVTdbjJjHXhCaocN6huWiks5qntJ4gaJpZM4Jz7yl>
> .
>